### PR TITLE
NonMR(SingleNode) PailConsolidation

### PR DESCRIPTION
**Pull request by ganeshpraburavi**
Have an edge that needs to be handled. Once you are in, We can review and merge.
@ashwanthkumar

### CODE REVIEW

**Comment by ganeshpraburavi** [comment]
@brewkode Please review and merge.
cc: @ashwanthkumar

File: dfs-datastores/src/test/java/com/backtype/hadoop/pail/PailOpsTest.java [hunk 1]
```diff
@@ -109,6 +109,24 @@ public void testConsolidationOne() throws Exception {
         assertEquals("lalala", pail.getMetadata("a/b/qqq"));
     }
 
+    public void testConsolidationSinglePath() throws IOException {
```
Comment (brewkode) [inline dfs-datastores/src/test/java/com/backtype/hadoop/pail/PailOpsTest.java]: better name `testConsolidateSinglePartition`

File: dfs-datastores/src/test/java/com/backtype/hadoop/pail/SingleTargetTestStructure.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.backtype.hadoop.pail;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SingleTargetTestStructure implements PailStructure<String> {
+
+    public boolean isValidTarget(String... dirs) {
+            return dirs != null & dirs.length >= 1;
```
Comment (brewkode) [inline dfs-datastores/src/test/java/com/backtype/hadoop/pail/SingleTargetTestStructure.java]: Pl use logical AND - &&

File: dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala [hunk 3]
```diff
@@ -9,7 +9,7 @@ import org.joda.time.DateTime
 import org.slf4j.LoggerFactory
 import util.{DateTimeFormatter, DateHelper}
 
-class PailConsolidate(inputDir: String, subDir: String, pipelineLabel: String)  {
+class PailConsolidate(inputDir: String, subDir: String, pipelineLabel: String, singleNodeConsolidation: Boolean = false)  {
```
Comment (brewkode) [inline dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala]: Pl use a parameter name that reflects the `nonMR` convention that's there in other places.

File: dfs-datastores/src/main/java/com/backtype/hadoop/Consolidator.java [hunk 4]
```diff
@@ -377,7 +401,7 @@ public long getValue() {
             return results;
         }
 
-        private String getParent(PailStructure<?> structure, Path path, String pailRoot) {
+        public String getParent(PailStructure<?> structure, Path path, String pailRoot) {
```
Comment (brewkode) [inline dfs-datastores/src/main/java/com/backtype/hadoop/Consolidator.java]: Better moved into the Utils class?

**Pull request merged**